SpatialGDK/Source/SpatialGDK/Public/Interop/RPCs/RPC_RingBufferWithACK_Receiver.h

```diff
@@ -102,15 +102,15 @@ class MonotonicRingBufferWithACKReceiver : public TRPCBufferReceiver<PayloadType
 		for (auto Iterator = this->ReceivedRPCs.CreateIterator(); Iterator; ++Iterator)
 		{
 			Worker_EntityId EntityId = Iterator->Key;
-			if (!CanExtract(EntityId))
-			{
-				continue;
-			}
 			ReceiverState& State = ReceiverStates.FindChecked(EntityId);
 			auto& Payloads = Iterator->Value;
 			uint32 ProcessedRPCs = 0;
 			for (const auto& Payload : Payloads)
 			{
+				if (!CanExtract(EntityId))
```

Comment:
Not super familiar with the logic but can we change this delete while iterate pattern into a 2 pass system where we deal with the removals in the 2nd pass. Seems fragile how it is currently, as evident from the bug.

Reply:
yeah I mistakenly thought the issue was with the iterator removal below, but I can see now from the additional context its more nuanced than that.

Reply:
Oh, I see, thank you very much for taking a look and suggesting this!